lib/source/pl/core/resolver.cpp

```diff
@@ -0,0 +1,41 @@
+#include <pl/core/resolver.hpp>
+
+using namespace pl::core;
+using namespace pl::api;
+using namespace pl::hlp;
+
+Result<Source *, std::string> Resolver::resolve(const std::string &path)  {
+    using result_t = Result<Source *, std::string>;
+    // look in cache
+    auto it = m_cachedSources.find(path);
+    if (it != m_cachedSources.end())
+        return result_t::good(&it->second);
+
+    Result<Source, std::string> result;
```

Comment:
It is for the reason that internal revolvers return objects, and the resolver manager keeps reference to them. This will be obsolete due to removing caching entirely